mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte

```diff
@@ -22,74 +25,51 @@
     },
   ];
 
-  let selectedTabId = 1;
-  $: selectedTab = tabs.find((tab) => tab.id === selectedTabId);
-
-  const handleTabClick = (tabId: number) => {
-    selectedTabId = tabId;
-  };
+  let activeTab: TabItem;
 </script>
 
 <div class="table-inspector-container">
-  <div class="mode-tabs-container">
-    {#each tabs as tab (tab.id)}
-      <span
-        class="mode-tab"
-        on:click={() => handleTabClick(tab.id)}
-        role="button"
-        class:is-selected={selectedTab?.id === tab.id}
-      >
-        {tab.label}
-      </span>
-    {/each}
-  </div>
-  {#if selectedTab}
-    <div class="tabs-container">
-      <svelte:component this={selectedTab.component} />
-    </div>
-  {/if}
+  <TabContainer bind:activeTab {tabs}>
+    <slot>
+      {#if activeTab}
+        <div class="tabs-container">
+          <svelte:component this={activeTab.component} />
+        </div>
+      {/if}
+    </slot>
+  </TabContainer>
 </div>
 
-<style>
+<style lang="scss">
   .table-inspector-container {
+    --collapsible-header-background-color: var(--sand-200);
     width: var(--table-inspector-width, 400px);
     box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.14),
       0px 3px 1px -2px rgba(0, 0, 0, 0.12), 0px 1px 5px 0px rgba(0, 0, 0, 0.2);
     position: relative;
+    background-color: var(--sand-100);
     isolation: isolate;
-  }
 
-  .tabs-container {
-    position: absolute;
-    overflow-y: auto;
-    left: 0;
-    right: 0;
-    bottom: 0;
-    top: 20px;
-    padding: 0.5rem;
-  }
+    :global(li.tab) {
+      flex: 1;
+    }
 
-  .mode-tabs-container {
-    display: flex;
-    flex-direction: row;
-    gap: 0.5rem;
-    z-index: 1;
-    position: relative;
-    background: white;
-    padding: 0.5rem;
-  }
+    :global(.tabs) {
+      margin-bottom: 0;
+    }
 
-  .mode-tab {
-    border: 1px solid rgba(0, 0, 0, 0.2);
-    border-radius: 4px;
-    padding: 0.15rem 0.25rem;
-    flex: 1;
-    text-align: center;
-    cursor: pointer;
-  }
+    :global(li.tab > div) {
+      text-align: center;
+      margin: auto;
+    }
 
-  .mode-tab:hover,
-  .mode-tab.is-selected {
-    background-color: rgba(0, 0, 0, 0.12);
+    :global(.tabs-container) {
+      left: 0;
+      right: 0;
+      bottom: 0;
+      top: 40px;
+      overflow-y: auto;
+      position: absolute;
+    }
   }
```

Comment:
All this global CSS makes me suspicious that the `<TabContainer>` component may not be designed well, though I haven't looked at it deeply enough to figure out what the issue is here. But this is to say, I wouldn't expect to see all this global CSS for a seemingly straightforward use-case of the tabs system. Feel free to disregard this comment if you've already given this some thought.

Reply:
I see there is still some global CSS in there now, but this is much better than before.